lib/default_cpolicy.json

```diff
@@ -0,0 +1,92 @@
+{
+	"rules": 
+	[
+		{
+			"name": "pln-rule-deduction",
+			"file_path": "reasoning/RuleEngine/rules/pln/deduction.scm",
+			"priority": 1,
+			"mutex": 
+			[
+				
+			],
+
+			"category": "PLN"
+		},
+
+		{
+			"name": "pln-rule-modus-ponens",
+			"file_path": "reasoning/RuleEngine/rules/pln/modus-ponens.scm",
+			"priority": 1,
+			"mutex": 
+			[
+				
+			],
+
+			"category": "PLN"
+		},
```

Comment:
Another very different and completely unrelated issue: a few weeks ago, there was a lengthly discussion about the relex2logic rules, which currently use priorities and mutually exclusive rule-sets.  There was considerable grumbling over this design, and I proposed a similar, but slightly different design.   I am somewhat disheartened to see that priority+mutex is showing up again, in new code.  It is likely to cause the same pain and heartache that R2L is complaining about, so I think it would be better to avoid it now.

Soooo... the slightly-different design is this.  Instead of priority, use "cost". A cost of zero is same as priority 1.  A cost of 1 is the same as priority 2. A cost of 0.5 is the same as a priority 1.5.  Costs are floating point.  Not important now, but will be very important later: costs are additive.  You will be able to add them together.

Next: instead of mutexes, use disjuncts. The disjuncts are stored separates from the rules, they are not part of a rule.  Instead, a collection of disjuncts look like this:
```
apply rule "modus ponens"
or apply rule "modus fashion"
or apply (rule "modus ice-cream" and rule "modus-pie")
or apply rule "modus fugit"
```
This gives you the choice of rules to apply at each stage, and they are, by definition mutually exclusive of each-other. That is what the "or" above is: its actually "exclusive-or".   Because each rule has a cost, you know what order the rules should be applied in.

This is a minor design change in the short run, and is almost completely identical to priority+mutex. However, in the long run, cost+disjunct is much more flexible and powerful and easier to use and manage.  So, please avoid priority+mutex.